### PR TITLE
Statistic plugin review

### PR DESCRIPTION
**Pull request by abolinbsu**

### CODE REVIEW

File: libraries/app/include/scorum/app/state.hpp [hunk 1]
```diff
@@ -153,7 +152,7 @@ FC_REFLECT( scorum::app::account_vote, (authorperm)(weight)(rshares)(percent)(ti
 
 FC_REFLECT( scorum::app::discussion_index, (category)(trending)(payout)(payout_comments)(trending30)(updated)(created)(responses)(active)(votes)(maturing)(best)(hot)(promoted)(cashout) )
 FC_REFLECT( scorum::app::tag_index, (trending) )
-FC_REFLECT_DERIVED( scorum::app::discussion, (scorum::app::comment_api_obj), (url)(root_title)(pending_payout_value)(total_pending_payout_value)(active_votes)(replies)(author_reputation)(promoted)(body_length)(reblogged_by)(first_reblogged_by)(first_reblogged_on) )
+FC_REFLECT_DERIVED( scorum::app::discussion, (scorum::app::comment_api_obj), (url)(root_title)(pending_payout_value)(active_votes)(replies)(author_reputation)(promoted)(body_length)(reblogged_by)(first_reblogged_by)(first_reblogged_on) )
```
Comment (Andrei-Masilevich) [inline libraries/app/include/scorum/app/state.hpp]: Try to use column order:

(url)
(root_title)
(pending_payout_value)
...

with '// clang-format off'

File: libraries/chain/evaluators/scorum_evaluators.cpp [hunk 2]
```diff
@@ -776,7 +773,7 @@ void withdraw_vesting_evaluator::do_apply(const withdraw_vesting_operation& o)
     {
         const auto& dprops = dprops_service.get();
         asset min_vests = asset(dprops.median_chain_props.account_creation_fee.amount, VESTS_SYMBOL);
-        min_vests.amount.value *= 10;
+        min_vests *= 10;
```
Comment (Andrei-Masilevich) [inline libraries/chain/evaluators/scorum_evaluators.cpp]: Move 10 to config.hpp. Discover why 10.

File: libraries/plugins/account_statistics/include/scorum/account_statistics/account_statistics_plugin.hpp [hunk 3]
```diff
@@ -154,17 +134,11 @@ FC_REFLECT(
...
-        vests_received_from_routes)(sbd_conversion_requests_created)(sbd_to_be_converted)(
-        sbd_conversion_requests_filled)(scorum_converted)(limit_orders_created)(limit_orders_filled)(
-        limit_orders_cancelled)(limit_order_scorum_paid)(limit_order_scorum_received)(limit_order_sbd_paid)(
-        limit_order_sbd_received)(total_pow)(estimated_hashpower))
-// SET_INDEX_TYPE( scorum::account_statistics::account_stats_bucket_object,)
+        author_rewards_vests)(author_rewards_total_scorum_value)(curation_reward_payouts)(curation_rewards_vests)(
+        curation_rewards_scorum_value)(transfers_to)(transfers_from)(scorum_sent)(scorum_received)(
+        transfers_to_vesting)(scorum_transferred_to_vesting)(new_vests)(new_vesting_withdrawal_requests)(
+        modified_vesting_withdrawal_requests)(vesting_withdrawals_processed)(finished_vesting_withdrawals)(
+        vests_withdrawn)(scorum_received_from_withdrawls)(scorum_received_from_routes)(vests_received_from_routes))
```
Comment (Andrei-Masilevich) [inline libraries/plugins/account_statistics/include/scorum/account_statistics/account_statistics_plugin.hpp]: Use column order

File: libraries/plugins/blockchain_statistics/blockchain_statistics_api.cpp [hunk 4]
```diff
@@ -64,7 +65,14 @@ statistics blockchain_statistics_api_impl::get_stats_for_interval(fc::time_point
 statistics blockchain_statistics_api_impl::get_lifetime_stats() const
 {
     statistics result;
-    result += _app.chain_database()->get(bucket_id_type());
+
+    const auto& bucket_idx = _app.chain_database()->get_index<bucket_index>().indices().get<by_bucket>();
+    auto itr = bucket_idx.find(boost::make_tuple(std::numeric_limits<uint32_t>::max(), fc::time_point_sec()));
```
Comment (Andrei-Masilevich) [inline libraries/plugins/blockchain_statistics/blockchain_statistics_api.cpp]: Use LIFE_TIME_PERIOD

File: tests/chain_tests/statistic/statistic_tests.cpp [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+    const bucket_object& bucket = get_lifetime_bucket();
+
+    auto orig_val = bucket.blocks;
+
+    generate_block();
+
+    BOOST_REQUIRE_EQUAL(bucket.blocks, orig_val + 1);
+}
+
+SCORUM_TEST_CASE(transactions_stat_test)
```
Comment (alexplevako) [inline tests/chain_tests/statistic/statistic_tests.cpp]: increase_transactions_counter_test

**Pull request merged**